v2/internal/x509util/util.go

```diff
@@ -0,0 +1,26 @@
+package x509util
+
+import (
+	"bytes"
+	"crypto/x509"
+)
+
+func NewCertPool(certs []*x509.Certificate) *x509.CertPool {
+	pool := x509.NewCertPool()
+	for _, cert := range certs {
+		pool.AddCert(cert)
+	}
+	return pool
+}
+
+// CertsEqual checks if two X.509 certificates are equal by comparing its raw bytes
+func CertsEqual(a, b *x509.Certificate) bool {
```

Comment:
I'm embarrassed I didn't remember this during the PR for the x509 bundle. This function is provided by the stdlib (x509.Certificate.Equal) and has the same behavior.